extensions/ql-vscode/src/data-extensions-editor/auto-model-usages-query.ts

```diff
@@ -0,0 +1,129 @@
+import { CancellationTokenSource } from "vscode";
+import { join } from "path";
+import { runQuery } from "./external-api-usage-query";
+import { CodeQLCliServer } from "../codeql-cli/cli";
+import { QueryRunner } from "../query-server";
+import { DatabaseItem } from "../databases/local-databases";
+import { interpretResultsSarif } from "../query-results";
+import { ProgressCallback } from "../common/vscode/progress";
+
+type Options = {
```

Comment:
This is okay for what we're doing, but the fact that we have to use `Pick` so much is a smell for me. It means entities aren't well broken down and it hurts readability.

Reply:
I've removed `Pick` from this definition. That means it's harder to test this part since you now need to know which methods are being called in the function rather than being able to derive that from the type, but it does improve readability.

Reply:
Yeah I appreciate it makes testing harder but I'm wary of using `Pick` and `Omit` as it can get out of hand quickly and really hurt readability. To me it's usually a sign that an interface is too big and needs to be broken down.

it may be that we'll decide to go with `Pick` in the short term anyway, but I'd rather we avoid it if possible as it spreads very easily.